### PR TITLE
Upgrade to java8

### PR DESCRIPTION
**Pull request by vladimir-bukhtoyarov**
This requests contains **minimal** set of changes which required to moving to java 8. List of upgrades:
1.   Java - from 6 to 8.
2.  Spring Framework from 3.1.4.RELEASE to 4.2.2.RELEASE
3. Hibernate - from 3.5.2.Final to 3.6.9.Final
4. JasperReports - from 4.0.1 to 6.1.1
5. Slf4j - from 1.5.2 to 1.7.5

The library versions are not the latest, because migration to fresh versions was not a goal.

### CODE REVIEW

**Comment by amikryukov** [comment]
:+1:

File: webclient/pom.xml [hunk 1]
```diff
@@ -217,10 +217,10 @@
         <plugins>
             <plugin>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>2.4</version>
+                <version>3.3</version>
                 <configuration>
-                    <target>1.6</target>
-                    <source>1.6</source>
+                    <target>1.8</target>
```
Comment (a-badaev) [inline webclient/pom.xml]: ${jagger.targetJdkVersion} should be used here

File: webclient/pom.xml [hunk 2]
```diff
@@ -217,10 +217,10 @@
         <plugins>
             <plugin>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>2.4</version>
+                <version>3.3</version>
```
Comment (a-badaev) [inline webclient/pom.xml]: plugin version is duplicated. Should be placed in the parent pom in pluginManagement section.

File: archetype-examples/src/main/resources/archetype-resources/pom.xml [hunk 3]
```diff
@@ -110,8 +110,8 @@
             <plugin>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>
-                    <source>1.6</source>
-                    <target>1.6</target>
+                    <source>1.8</source>
```
Comment (a-badaev) [inline archetype-examples/src/main/resources/archetype-resources/pom.xml]: ${jagger.targetJdkVersion} should be used here

File: pom.xml [hunk 4]
```diff
@@ -242,27 +242,20 @@
...
                 <artifactId>hibernate-core</artifactId>
-                <version>3.5.4-Final</version>
-                <scope>compile</scope>
-            </dependency>
-
-            <dependency>
-                <groupId>org.hibernate</groupId>
-                <artifactId>hibernate-annotations</artifactId>
-                <version>3.5.4-Final</version>
+                <version>3.6.9.Final</version>
```
Comment (a-badaev) [inline pom.xml]: hibernate version should be moved to properties section

File: pom.xml [hunk 5]
```diff
@@ -242,27 +242,20 @@
...
                 <scope>compile</scope>
             </dependency>
 
             <dependency>
                 <groupId>org.hibernate</groupId>
                 <artifactId>hibernate-entitymanager</artifactId>
-                <version>3.5.4-Final</version>
+                <version>3.6.9.Final</version>
```
Comment (a-badaev) [inline pom.xml]: hibernate version should be moved to properties section

File: pom.xml [hunk 6]
```diff
@@ -24,18 +24,18 @@
...
     <url>http://maven.apache.org</url>
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <jagger.targetJdkVersion>1.6</jagger.targetJdkVersion>
+        <jagger.targetJdkVersion>1.8</jagger.targetJdkVersion>
         <jagger.targetJdk>${JAVA_HOME}</jagger.targetJdk>
 
-        <spring.version>3.1.4.RELEASE</spring.version>
+        <spring.version>4.2.2.RELEASE</spring.version>
```
Comment (a-badaev) [inline pom.xml]: Why it is not the latest - 4.3.2.RELEASE ?

**Comment by a-badaev** [comment]
Project version should not be updated. It can only be updated automatically by Jenkins. And I think  hibernate.validator.version could be updated to 5.2.4.Final

**Comment by kgribov** [comment]
Guys, highly recommend to make small PRs, one feature - one PR.
Because now it is really hard to review PR with 44 files. 
Here I can see, there are 5 changes, one with java update and four with frameworks updates. So you could create 5 different PRs. This is doesn't mean to close this PR or refactor it, it is just a good practice, which we could follow. 
Smaller PRs -> faster review.
Thanks

**Comment by kgribov** [comment]
👍

File: pom.xml (hunk 5 above)
Comment (vladimir-bukhtoyarov) [inline pom.xml]: Updating to last versions was not a my goal, I tried to do minimal changes which enough to migrate to java8

**Comment by vladimir-bukhtoyarov** [comment]
The splitting to several requests has no sense because build will will be broken.

**Comment by vladimir-bukhtoyarov** [comment]
> Project version should not be updated. It can only be updated automatically by Jenkins.
Ok, I will revert artifact versions.

> And I think hibernate.validator.version could be updated to 5.2.4.Final
Updating to latest versions of thirdparty libraries ws not a goal, I tried to do minimal set of changes which required to upgrade to java 8.

So, I am going to provide PR today

**Comment by kgribov** [comment]
@vladimir-bukhtoyarov 
> The splitting to several requests has no sense because build will will be broken.

Why?

**Comment by nmusienko** [comment]
:open_mouth:

**Pull request merged**